build-tools/src/main/resources/software/amazon/awssdk/checkstyle.xml

```diff
@@ -345,10 +345,20 @@
         <module name="Regexp">
           <property name="format" value="System\s*(\.|::)\s*(getenv|getProperty)"/>
           <property name="illegalPattern" value="true"/>
-          <property name="message" value="Use SystemSetting instead of System.getenv and System.getProperty"/>
+          <property name="message" value="NEVER use System.getenv or System.getProperty. Create and use a
+          SystemSetting, instead."/>
           <property name="ignoreComments" value="true"/>
         </module>
 
+        <module name="Regexp">
+            <property name="format" value="SystemSetting\s*(\.|::)\s*getStringValueFromEnvironmentVariable"/>
+            <property name="illegalPattern" value="true"/>
+            <property name="message" value="Are you being naughty and not creating a SystemSetting? Don't you know that
+            your customers want both system properties AND environment variables? Are you REALLY sure you want to use
+            this and not support a system property?"/>
+            <property name="ignoreComments" value="true"/>
+        </module>
```

Comment:
Note for reviewers: This was modified to really push people to use SystemSetting, because that's the only mechanism that supports environment variable mocking, now. We expect everyone to create new SystemSettings to support both properties and environment variables. If they really want to break that, then they can use the static `SystemSetting.getStringValueFromEnvironmentVariable` (long to be annoying to write), and they still have to add a checkstyle suppression to do it.